Make Ghost Editor compatible with showdown latest

tivie opened the pull request:
- turned a bunch of options on in showdown to set a "ghost" flavor markdown, similar to GFM
 - custom extensions live in their own repo, with proper tests against showdown
 - updated dependencies in package.json
 - should fix a number of issues and bugs in markdown parsing

tivie committed:
Make Ghost Editor compatible with showdown latest

 - changed showdown global
 - turned a bunch of options on in showdown to set a "ghost" flavor markdown, similar to GFM
 - custom extensions live in their own repo, with proper tests against showdown
 - updated dependencies in package.json

tivie commented:
Should you choose to accept this PR, I can change ownership of the custom extensions repos to you.

disordinary commented:
Thanks for that @tivie,

This will be a welcome improvement in our markdown rendering!

tivie commented:
No problem. You should decide what showdown options should be turned on and off so that I can create, in showdown, a "Ghost Flavor".

This way, code would be simplified and any changes in showdown via options could be integrated, automatically, in Ghost.

tivie committed:
Update showdown and showdown-extensions versions in package.json

 - Uses showdown ^1.6.3
 - Uses ^0.0.x for custom ghost extensions.
 - Latests versions fix some bugs and edge issues that were mentioned in issues or tested in Ghost repository

tivie commented:
Added one more commit to bump package.json versions.

Both showdown and extensions were tested in Ghost environment against a bigger number of tests.
As a result, some minor issues and edge cases were fixed.
